extensions/applicationinsights-cfgsync-js/src/CfgSyncPlugin.ts

```diff
@@ -0,0 +1,412 @@
+                        let JSON = getJSON();
+                        if (JSON) {
+                            let cfg = JSON.parse(response); //comments are not allowed
+                            cfg && _setCfg(cfg, isAutoSync);
+                        }
+                    } else {
+                        _retryCnt ++;
+                    }
+                } catch (e) {
+                    // eslint-disable-next-line no-empty
+                }
+            }
+
+            function _doOnComplete(oncomplete: OnCompleteCallback, status: number, response?: string, isAutoSync?: boolean) {
+                try {
+                    oncomplete(status, response, isAutoSync);
+                } catch (e) {
+                    // eslint-disable-next-line no-empty
+                }
+            }
+
+            function _addEventListener() {
+                if (_receiveChanges) {
+                    let global = getGlobal();
+                    if (global) {
+                        try {
+                            eventOn(global, _evtName, (event: any) => {
+                                let cfgEvent = event && (event as any).detail;
+                                if (_onCfgChangeReceive && cfgEvent) {
+                                    _onCfgChangeReceive(cfgEvent);
+                                } else {
+                                    let cfg = cfgEvent && cfgEvent.cfg;
+                                    let newCfg = cfg && _replaceTartgetByKeys(cfg);
+                                    newCfg && _self.core.updateCfg(newCfg);
+                                }
+                            }, _evtNamespace, true);
+
+                        } catch(e) {
+                            // eslint-disable-next-line no-empty
+                        }
+                    }
+                }
+            }
+            // 4 levels
+            function _replaceTartgetByKeys<T=IConfiguration & IConfig>(cfg: T , level?: number) {
+                let _cfg: IConfiguration & IConfig = null;
+                try {
+                    if (cfg) {
+                        _cfg = replaceByNonOverrideCfg(cfg, _nonOverrideConfigs, 0, 5);
+                    }
+                } catch(e) {
+                    // eslint-disable-next-line no-empty
+                }
+                return _cfg;
+            }
+
+            /**
+             * Sets up the timer which triggers fetching cdn every 30mins after inital call
+             */
+            function _setupTimer() {
+                if (!_timeoutHandle && _fetchSpan) {
+                    _timeoutHandle = scheduleTimeout(() => {
+                        _timeoutHandle = null;
+                        _fetchFn(_cfgUrl, _onFetchComplete, _broadcastChanges);
+                        if (_retryCnt < 3) {
+                            _setupTimer(); // only set new timer when retry count < 3
```

Comment:
We should probably do this in the `_onFetchComplete` function so that we don't try to send multiple requests at the same time (assuming one previous request takes a long time and someone has set the timeout to a small value)